zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SparkSQLGeneratorTest.scala

```diff
@@ -0,0 +1,167 @@
+          |(select *
+          |from twitter.ds_tweet t);""".stripMargin)
+    }
+    //self query test 8
+    "translate get min field value query without group by" in {
+      val globalAggr = GlobalAggregateStatement(aggrMin)
+      val query = new Query(dataset = TwitterDataSet, globalAggr = Some(globalAggr))
+      val result = parser.generate(query, Map(TwitterDataSet -> twitterSchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """select min(t.`id`) as `min` from
+          |(select *
+          |from twitter.ds_tweet t);""".stripMargin)
+    }
+    //self query test 9
+    "translate get max field value query without group by" in {
+      val globalAggr = GlobalAggregateStatement(aggrMax)
+      val query = new Query(dataset = TwitterDataSet, globalAggr = Some(globalAggr))
+      val result = parser.generate(query, Map(TwitterDataSet -> twitterSchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """select max(t.`id`) as `max` from
+          |(select *
+          |from twitter.ds_tweet t);""".stripMargin)
+    }
+    //self query test 10
+    "translate a count cardinality query with filter without group by" in {
+      val filter = Seq(timeFilter)
+      val globalAggr = GlobalAggregateStatement(aggrCount)
+      val query = new Query(dataset = TwitterDataSet, filter = filter, globalAggr = Some(globalAggr))
+      val result = parser.generate(query, Map(TwitterDataSet -> twitterSchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """select count(*) as `count` from
+          |(select *
+          |from twitter.ds_tweet t
+          |where t.`create_at` >= '2016-01-01T00:00:00.000Z' and t.`create_at` < '2016-12-01T00:00:00.000Z');""".stripMargin)
+    }
+    //self query test 11
+    "translate a min cardinality query with filter without group by" in {
+      val filter = Seq(timeFilter)
+      val globalAggr = GlobalAggregateStatement(aggrMin)
+      val query = new Query(dataset = TwitterDataSet, filter = filter, globalAggr = Some(globalAggr))
+      val result = parser.generate(query, Map(TwitterDataSet -> twitterSchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """select min(t.`id`) as `min` from
+          |(select *
+          |from twitter.ds_tweet t
+          |where t.`create_at` >= '2016-01-01T00:00:00.000Z' and t.`create_at` < '2016-12-01T00:00:00.000Z');""".stripMargin)
+    }
+    //self query test 12
+    "translate a count cardinality query with select" in {
+      val globalAggr = GlobalAggregateStatement(aggrCount)
+      val query = new Query(dataset = TwitterDataSet, select = Some(selectTop10), globalAggr = Some(globalAggr))
+      val result = parser.generate(query, Map(TwitterDataSet -> twitterSchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """select count(*) as `count` from
+          |(select *
+          |from twitter.ds_tweet t
+          |limit 10);""".stripMargin.trim)
+    }
+
```

Comment:
Missing too many test cases. I think you should convert virtually all test cases existing for SQL++ for your generator.